plugins/kubernetes/src/components/Terminal/Terminal.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import 'xterm/css/xterm.css';
+
+import { configApiRef, useApi } from '@backstage/core-plugin-api';
+import { IContainerStatus } from 'kubernetes-models/v1';
+import React, { useEffect, useMemo } from 'react';
+import { Terminal } from 'xterm';
+import { FitAddon } from 'xterm-addon-fit';
+
+import { PodScope } from '../Pods';
+import { K8sAttachAddon } from './K8sAttachAddon';
+
+export interface TerminalProps {
+  podScope: PodScope;
+  containerStatus: IContainerStatus;
+}
+
+export const K8sTerminal = (props: TerminalProps) => {
+  const { podScope, containerStatus } = props;
+
+  const terminalRef = React.useRef(null);
+  const configApi = useApi(configApiRef);
+  const namespace = podScope.podNamespace ?? 'default';
+
+  const baseUrl = useMemo(() => {
+    let url =
+      configApi.getOptionalString('backend.baseUrl') ?? window.location.host;
+
+    url = url.replace(/^https?:\/\//, '');
+
+    return url;
+  }, [configApi]);
+
+  const urlParams = useMemo(() => {
+    const params = new URLSearchParams({
+      container: containerStatus.name,
+      stdin: 'true',
+      stdout: 'true',
+      stderr: 'true',
+      tty: 'true',
+      command: '/bin/bash',
+    });
+    return params;
+  }, [containerStatus.name]);
+
+  const socketUrl = useMemo(
+    () =>
+      new URL(
+        `ws://${baseUrl}/api/kubernetes/proxy/api/v1/namespaces/${namespace}/pods/${podScope.podName}/exec?${urlParams}`,
```

Comment:
The TLS-enabled version of the websocket scheme is `wss` -- i.e. if `baseUrl` starts with `https://`, then `socketUrl` should start with `wss://` instead of `ws://`

Reply:
Moved the protocol part up to the baseUrl memo, that should take care of both cases.